rfcs/text/0015_bazel.md

```diff
@@ -0,0 +1,308 @@
+
+Example package BUILD.bazel for `packages/elastic-datemath`:
+
+```python
+load("@build_bazel_rules_nodejs//:index.bzl", "pkg_npm")
+load("@build_bazel_rules_nodejs//internal/js_library:js_library.bzl", "js_library")
+
+SRCS = [
+    ".npmignore",
+    "index.js",
+    "index.d.ts",
+    "package.json",
+    "readme",
+    "tsconfig.json",
+]
+
+filegroup(
+    name = "src",
+    srcs = glob(SRCS),
+)
+
+js_library(
+    name = "elastic-datemath",
+    srcs = [ ":src" ],
+    deps = [ "@npm//moment" ],
+    package_name = "@elastic/datemath",
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "build",
+    actual = "elastic-datemath",
+    visibility = ["//visibility:public"],
+)
+```
+
+If the package has unit tests, they will need to be migrated which will be invoked with `bazel test` as described in the Unit Testing section.
+
+
+## Phase II - Docs, Developer Experience
+
+Packages were a likely choice for phase 1 for a few reasons; they aren’t often updated and the developer experience is quite lacking making it easy to maintain parity with. In phase 2, we will bring the developer experience of packages to that which developers are accustomed to with plugins. This means re-builds will be automatic when a change occurs as well as giving time to address any developer experience shortcomings which were not foreseen. During this time we will work on overall Bazel documentation as it pertains to the Kibana repository.
+
+
+## Phase III - Plugins
+
+In this phase, we will be migrating each of the 135 plugins over to being built and unit tested using Bazel. During this time, the legacy systems will stay in place and run in parallel with Bazel. Once all plugins have been migrated, we can decommission the legacy systems.
+
+The `BUILD.bazel` files will look similar to that of packages, there will be a target for `web`, `server`, and `jest`. Just like packages, as the Jest unit tests are migrated, they will need to be removed from the root `jest.config.js` file as described in the Unit Testing section.
+
+Plugins are built in a sandbox, so they will no longer be able to use relative imports from one another. For Typescript, relative imports will be replaced with a path reference to the `bazel/bin`.
+
+Static imports across plugins are a concern that would affect the developer experience due to cascading re-builds. For example, if every plugin has static imports from `src/core`, any changes to `src/core` would cause all those plugins to re-build. There are a few options to address this; the first would be to minimize or eliminate these imports. Most plugins are importing types, so we can also ensure that only type-level changes actually trigger a re-build. Additionally, these types of dependencies could be further broken out into another package to reduce the times further this is necessary. 
```

Comment:
Is Bazel able to understand whether the code or type has been changed? Or we need to annotate import/export explicitly with `import type { ... } from 'src/core/server'`

Reply:
For the best of our knowledge today it looks like the typescript rules for Bazel only triggers cascading rebuilds when the types changed. On the tests I did with @tylersmalley we were not using `import type`.

Reply:
I think this is a bit more clear, if I'm understanding correctly.

```suggestion
Static imports across plugins are a concern that would affect the developer experience due to cascading re-builds. For example, if every plugin has static imports from `src/core`, any changes to `src/core` would cause all those plugins to re-build. There are a few options to address this; the first would be to minimize or eliminate these imports. Most plugins are importing types, so we can also ensure that only type-level changes actually trigger a re-build. Additionally, these types of dependencies could be further broken down into smaller packages to reduce the times further this is necessary. 
```